ics_space_amplification_test.py

```diff
@@ -0,0 +1,192 @@
+
+    def _get_max_used_capacity_over_time_gb(self, node, start_time):
+        """
+
+        :param node:
+        :param start_time: the start interval to search max-used-capacity from.
+        :return:
+        """
+        fs_size_gb = self._get_filesystem_total_size_gb(node=node)
+        end_time = time.time()
+        time_interval_minutes = int(math.ceil((end_time - start_time) / 60))  # convert time to minutes and round up.
+        min_available_capacity_gb = min([int(val[1]) for val in self._get_filesystem_available_size_list(node=node,
+                                                                                                         start_time=start_time)]) / GB_SIZE
+        max_used_capacity_gb = fs_size_gb - min_available_capacity_gb
+        self.log.debug("The maximum used filesystem capacity of {} for the last {} minutes is: {} GB/ {} GB".format(
+            node.private_ip_address, time_interval_minutes, max_used_capacity_gb, fs_size_gb))
+        return max_used_capacity_gb
+
+    def _get_nodes_space_amplification(self, dict_nodes_initial_capacity, start_time) -> dict:
+        dict_nodes_space_amplification = {}
+        dict_nodes_used_capacity = self._get_nodes_used_capacity()
+        for node in self.db_cluster.nodes:
+            node_max_used_capacity_gb = self._get_max_used_capacity_over_time_gb(node=node,
+                                                                                 start_time=start_time)
+            dict_nodes_space_amplification[node.private_ip_address] = node_max_used_capacity_gb - \
+                dict_nodes_initial_capacity[
+                node.private_ip_address]
+            self.log.info(
+                f"Node {node.private_ip_address} used capacity changed from {dict_nodes_initial_capacity[node.private_ip_address]} to {dict_nodes_used_capacity[node.private_ip_address]}.")
+            self.log.info(f"Space amplification is: {dict_nodes_space_amplification[node.private_ip_address]} GB")
+        return dict_nodes_space_amplification
+
+    def _get_nodes_space_amplification_after_write(self, dict_nodes_initial_capacity, written_data_size_gb,
+                                                   start_time) -> dict:
+        self.log.info(f"Space amplification results after a write of: {written_data_size_gb} are:")
+        return self._get_nodes_space_amplification(dict_nodes_initial_capacity=dict_nodes_initial_capacity,
+                                                   start_time=start_time)
+
+    def _get_nodes_used_capacity(self) -> dict:
+        """
+
+        :rtype: dictionary with capacity per node-ip
+        """
+        dict_nodes_used_capacity = {}
+        for node in self.db_cluster.nodes:
+            dict_nodes_used_capacity[node.private_ip_address] = self._get_used_capacity_gb(node=node)
+        return dict_nodes_used_capacity
+
+    @retrying(n=80, sleep_time=60, allowed_exceptions=(AssertionError,))
+    def wait_no_compactions_running(self):
```

Comment:
how this is different from `wait_no_compactions_running()` in `tester.py`

Reply:
it is different in: (1) n=80  (2) "if results: assert..."

Reply:
that's not enough diffrence to justify two function doing the same thing, and with the same name.
refactor to function if needed to add needed functionality from it. or at least warp it with with needed code.

Reply:
fixed in tester.py